### ISSUE TITLE
Crashes on Julia 1.6

### ISSUE THREAD

**Issue by lassepe** [fannheyward/coc-julia#151]
Julia 1.6 has just been released. While the vscode plugin seems to work with Julia 1.6, `coc-julia` crashes.

**Comment by wangl-cc** [comment]
The coc-julia works for me with Julia v1.6 official release. If you have compiled LS, you may need to  compile agin for new version.

**Comment by lassepe** [comment]
How can I trigger re-compilation of the language server? I think it did that automatically when I first installed it. However, removing the `coc-julia` extension and adding it again does not trigger compilation.

**Comment by fannheyward** [comment]
Try `:CocCommand julia.CompileLanguageServerSysimg` again?

**Comment by lassepe** [comment]
I think there must be another issue. When running `:CocCommand julia.CompileLanguageServerSysimg` then I get the following error:

```julia
[coc.nvim]: Error on notification "runCommand": Command failed: /home/lassepe/.asdf/shims/julia --project=/home/lassepe/.config/coc/extensio
ns/node_modules/coc-julia/server/JuliaLS --startup-file=no --history-file=no -e "import LanguageServer; println(joinpath(dirname(pathof(Lang
uageServer)), \"../test/runtests.jl\"))"
ERROR: LoadError: KeyError: key :_apply_latest not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:482 [inlined]
  [2] getindex
    @ ~/.julia/packages/SymbolServer/eITqF/src/symbols.jl:20 [inlined]
  [3] load_core()
    @ SymbolServer ~/.julia/packages/SymbolServer/eITqF/src/symbols.jl:437
  [4] top-level scope
    @ ~/.julia/packages/SymbolServer/eITqF/src/SymbolServer.jl:203
  [5] include
    @ ./Base.jl:386 [inlined]
  [6] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector
{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1213
  [7] top-level scope
    @ none:1
  [8] eval
    @ ./boot.jl:360 [inlined]
  [9] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [10] top-level scope
    @ none:1
in expression starting at /home/lassepe/.julia/packages/SymbolServer/eITqF/src/SymbolServer.jl:1
ERROR: LoadError: Failed to precompile SymbolServer [cf896787-08d5-524d-9de7-132aaa0cb996] to /home/lassepe/.julia/compiled/v1.6/SymbolServe
r/jl_OjU9P3.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::Base.PipeEndpoint, internal_stdout::Base.PipeEndpoint)
    @ Base ./loading.jl:1360
  [3] compilecache(pkg::Base.PkgId, path::String)
    @ Base ./loading.jl:1306
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1021
  [5] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:914
  [6] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:901
  [7] include
    @ ./Base.jl:386 [inlined]
  [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector
{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1213
  [9] top-level scope
    @ none:1
 [10] eval
    @ ./boot.jl:360 [inlined]
 [11] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [12] top-level scope
    @ none:1
in expression starting at /home/lassepe/.julia/packages/LanguageServer/KQgOJ/src/LanguageServer.jl:1
ERROR: Failed to precompile LanguageServer [2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7] to /home/lassepe/.julia/compiled/v1.6/LanguageServer/jl_1F
e64G.
```

**Comment by lassepe** [comment]
Also, I should say that  if system image precompilation requires to manually run the command above, then I certainly never did that before. Hence, the crashes seem not to be related to the precompilation. Unfortunately, the output of `CocInfo` is not very informative:

```
2021-03-26T10:20:15.060 INFO (pid:3271154) [services] - registered service "julia"
2021-03-26T10:20:15.060 INFO (pid:3271154) [services] - Julia Language Server state change: stopped => starting
2021-03-26T10:20:15.065 INFO (pid:3271154) [language-client-index] - Language server "julia" started with 3271955
2021-03-26T10:20:21.345 INFO (pid:3271154) [attach] - receive notification: highlight []
2021-03-26T10:20:28.901 INFO (pid:3271154) [services] - Julia Language Server state change: starting => stopped
2021-03-26T10:20:28.901 INFO (pid:3271154) [services] - Julia Language Server state change: stopped => starting
2021-03-26T10:20:28.904 INFO (pid:3271154) [language-client-index] - Language server "julia" started with 3273202
2021-03-26T10:20:29.256 INFO (pid:3271154) [attach] - receive notification: highlight []
2021-03-26T10:20:44.007 INFO (pid:3271154) [services] - Julia Language Server state change: starting => stopped
2021-03-26T10:20:44.007 INFO (pid:3271154) [services] - Julia Language Server state change: stopped => starting
2021-03-26T10:20:44.016 INFO (pid:3271154) [language-client-index] - Language server "julia" started with 3274596
2021-03-26T10:20:58.572 INFO (pid:3271154) [services] - Julia Language Server state change: starting => stopped
2021-03-26T10:20:58.572 INFO (pid:3271154) [services] - Julia Language Server state change: stopped => starting
2021-03-26T10:20:58.578 INFO (pid:3271154) [language-client-index] - Language server "julia" started with 3275997
2021-03-26T10:21:11.516 INFO (pid:3271154) [services] - Julia Language Server state change: starting => stopped
2021-03-26T10:21:11.516 INFO (pid:3271154) [services] - Julia Language Server state change: stopped => starting
2021-03-26T10:21:11.525 INFO (pid:3271154) [language-client-index] - Language server "julia" started with 3277270
2021-03-26T10:21:25.762 INFO (pid:3271154) [services] - Julia Language Server state change: starting => stopped
2021-03-26T10:21:59.128 INFO (pid:3271154) [attach] - receive notification: showInfo []
```

**Comment by fannheyward** [comment]
I'll test the precompilation later.

**Comment by lassepe** [comment]
Maybe another relevant data point: I just deleted `~/.julia` and `~/.config/coc` to make sure I have a clean environment. Then at the first startup I get the prompts to install `compile_env` and `JuliaLS`. during the setup of the latter, the precompilation of some of the packages fails:

```
╭─lassepe@lassepe-x1e ~/worktree
╰─➤  /home/lassepe/.asdf/shims/julia --project="/home/lassepe/.config/coc/extensions/node_modules/coc-julia/server/JuliaLS" --startup-file=no --history-file=no -e "using Pkg; Pkg.instantiate()"
     Cloning git-repo `https://github.com/julia-vscode/LanguageServer.jl`
   Installed JSONRPC ──────── v1.2.0
   Installed Tokenize ─────── v0.5.8
   Installed URIParser ────── v0.4.1
   Installed Parsers ──────── v1.0.15
   Installed DocumentFormat ─ v3.2.0
   Installed FilePathsBase ── v0.9.6
   Installed JSON ─────────── v0.21.1
   Installed SymbolServer ─── v5.1.0
   Installed StaticLint ───── v4.5.0
   Installed CSTParser ────── v2.5.0
Precompiling project...

  ✗ SymbolServer
  ✗ StaticLint
  ✗ LanguageServer
  ✗ JuliaLS
8 dependencies successfully precompiled in 61 seconds
4 dependencies errored
```

**Comment by lassepe** [comment]
I think I found the issue. Apparently the `Manifest.toml` that is downloaded to `~/.config/coc/extensions/node_modules/coc-julia/server/JuliaLS/Manifest.toml` is corrupted. When I delete it and add `LanguageServer#master` manually in package mode then things seem to work.

**Comment by wangl-cc** [comment]
Depends of JuliaLS might be out of date. I have updated several days ago to test. Try to update it?

**Comment by lassepe** [comment]
Yes. Removing manually running `pkg> dev LanguageServer#master` in the environment of `~/.config/coc/extensions/node_modules/coc_julia/server/JuliaLS` fixed the issue. Maybe the problem was that it was set to `#master` and master is a moving target. Maybe it would be better to set it to a specific commit id or version in this package. Or is this not desired because then this repo would need to be updated a lot.

**Comment by RGerzaguet** [comment]
> Maybe another relevant data point: I just deleted `~/.julia` and `~/.config/coc` to make sure I have a clean environment. Then at the first startup I get the prompts to install `compile_env` and `JuliaLS`. during the setup of the latter, the precompilation of some of the packages fails:
> 
> ```
> ╭─lassepe@lassepe-x1e ~/worktree
> ╰─➤  /home/lassepe/.asdf/shims/julia --project="/home/lassepe/.config/coc/extensions/node_modules/coc-julia/server/JuliaLS" --startup-file=no --history-file=no -e "using Pkg; Pkg.instantiate()"
>      Cloning git-repo `https://github.com/julia-vscode/LanguageServer.jl`
>    Installed JSONRPC ──────── v1.2.0
>    Installed Tokenize ─────── v0.5.8
>    Installed URIParser ────── v0.4.1
>    Installed Parsers ──────── v1.0.15
>    Installed DocumentFormat ─ v3.2.0
>    Installed FilePathsBase ── v0.9.6
>    Installed JSON ─────────── v0.21.1
>    Installed SymbolServer ─── v5.1.0
>    Installed StaticLint ───── v4.5.0
>    Installed CSTParser ────── v2.5.0
> Precompiling project...
> 
>   ✗ SymbolServer
>   ✗ StaticLint
>   ✗ LanguageServer
>   ✗ JuliaLS
> 8 dependencies successfully precompiled in 61 seconds
> 4 dependencies errored
> ```
Hello,
I get exactly the same issue on fresh Julia 1.6 install for the pre-compilation. Not really sure to know how to handle that though. Any straightforward way ?

**Comment by lassepe** [comment]
Hello @RGerzaguet,

See my last comment:

> Manually running `pkg> dev LanguageServer#master` in the environment of `~/.config/coc/extensions/node_modules/coc_julia/server/JuliaLS` fixed the issue.

**Comment by RGerzaguet** [comment]
> dev LanguageServer#master

Yep it solves the problem, many thanks :)
And I see that @wangl-cc creates an issue to solve the problem as its roots, so this perfect !
Thanks again

**Comment by fannheyward** [comment]
I've merged #153 and published a new release, v0.8.0. Please give it a test, thank you guys for the tests and fixes.

**Comment by carlocab** [comment]
FYI, I tried this with the latest version of `coc-julia`, and I think I encountered the same issue described here. In particular `:CocInfo` showed me the same thing as in https://github.com/fannheyward/coc-julia/issues/151#issuecomment-808065479.

Running
```
julia --project="~/.config/coc/extensions/node_modules/coc-julia/server/JuliaLS" --startup-file=no --history-file=no -e "using Pkg; Pkg.instantiate()"
```
fixed it.
